### PR TITLE
Crm bidding strategies

### PR DESCRIPTION
**Pull request by nick-harder**

### CODE REVIEW

File: assume/common/utils.py [hunk 1]
```diff
@@ -291,6 +289,8 @@ def aggregate_step_amount(orderbook: Orderbook, begin=None, end=None, groupby=[]
 
     If called without groupby, this returns the aggregated orderbook timeseries
     """
+    if groupby is None:
+        groupby = []
```
Comment (maurerle) [inline assume/common/utils.py:293]: What is the benefit of this?

File: assume/common/utils.py [hunk 2]
```diff
@@ -316,11 +320,16 @@ def aggregate_step_amount(orderbook: Orderbook, begin=None, end=None, groupby=[]
...
-                    (date + timedelta(hours=duration_hours), -bid["volume"]) + add
+                deltas.extend(
+                    (
+                        (date, bid["volume"]) + add,
+                        (
+                            date + timedelta(hours=duration_hours),
+                            -bid["volume"],
+                        )
+                        + add,
+                    )
```
Comment (maurerle) [inline assume/common/utils.py:331]: if you want to reduce appends you should also list comprehend the for loop `for date in starts:`.

Well I don't know - thats probably not our main bottleneck is it?

File: assume/common/utils.py (hunk 1 above)
Comment (nick-harder) [reply assume/common/utils.py:293]: before the groupby was  mutable argument, which is not desired. so I replaced it but this is the work around

File: assume/common/utils.py (hunk 2 above)
Comment (nick-harder) [reply assume/common/utils.py:331]: yeah, could be helpful as well. Maybe not a bottleneck, but some optimization and nices looking code is still nice.

**Pull request merged**